src/actions/router.js

```diff
@@ -49,7 +97,7 @@ export function goToMessage(message) {
       type: types.GO_TO_MESSAGE,
       payload: message
     })
-    dispatch(goTo({path: parseUrl(message.link).pathname}))
+    dispatch(goTo({path: parseUrl(message.link).href}))
```

Comment:
1. there is no need to parse if you take the href, you can take the message.link directly then
2. it is not a path in that case, it is a url
3. I see that we need to refactor goTo logic as well. We need a new interface goTo(pathOrUrl, options). You will have to detect if its  a path or a full url.